KafkaIngester/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -303,7 +313,75 @@ private Json(
         }
     }
 
-    // Spec to explicitly ask consumeToTabel to ignore either key or value.
+    /**
+     * Type enumeration for the result {@link Table} returned by stream consumers.
+     */
+    public enum TableType {
+        /**
+         * <p>Consume all partitions into a single interleaved stream table, which will present only newly-available
+         * rows to downstream operations and visualizations.
+         * <p>See {@link Table#STREAM_TABLE_ATTRIBUTE} for a detailed explanation of stream table semantics, and
+         * {@link io.deephaven.db.v2.StreamTableTools} for related tooling.
+         */
+        Stream(false, false),
+        /**
+         * Consume all partitions into a single interleaved in-memory append-only table.
+         */
+        Append(true, false),
+        /**
+         * <p>As in {@link #Stream}, but each partition is mapped to a distinct stream table.
+         * <p>The resulting per-partition tables are aggregated into a single {@link TableMap} keyed by
+         * {@link Integer} partition, which is then presented as a {@link Table} proxy via
+         * {@link TransformableTableMap#asTable(boolean, boolean, boolean) asTable} with {@code strictKeys=true},
+         * {@code allowCoalesce=true}, and {@code sanityCheckJoins=true}.
+         * <p>See {@link TransformableTableMap#asTableMap()} to explicitly work with the underlying {@link TableMap}
+         * and {@link TransformableTableMap#asTable(boolean, boolean, boolean)} for alternative proxy options.
+         */
+        StreamMap(false, true),
+        /**
+         * <p>As in {@link #Append}, but each partition is mapped to a distinct in-memory append-only table.
+         * <p>The resulting per-partition tables are aggregated into a single {@link TableMap} keyed by
+         * {@link Integer} partition, which is then presented as a {@link Table} proxy via
+         * {@link TransformableTableMap#asTable(boolean, boolean, boolean) asTable} with {@code strictKeys=true},
+         * {@code allowCoalesce=true}, and {@code sanityCheckJoins=true}.
+         * <p>See {@link TransformableTableMap#asTableMap()} to explicitly work with the underlying {@link TableMap}
+         * and {@link TransformableTableMap#asTable(boolean, boolean, boolean)} for alternative proxy options.
+         */
+        AppendMap(true, true);
+
+        private final boolean isAppend;
+        private final boolean isMap;
+
+        TableType(final boolean isAppend, final boolean isMap) {
+            this.isAppend = isAppend;
+            this.isMap = isMap;
+        }
+
+    }
+
+    /**
+     * Map "Python-friendly" table type name to a {@link TableType}.
+     *
+     * @param typeName The friendly name
+     * @return The mapped {@link TableType}
+     */
+    public static TableType friendlyNameToTableType(@NotNull final String typeName) {
+        // @formatter:off
+        switch (typeName) {
+            case "streaming"    : return TableType.Stream;
```

Comment:
Should we change these to "stream" to match the TableType and also "StreamTable" as opposed to "StreamingTable".

Reply:
I think yes, if @jcferretti agrees.

Reply:
You native and educated english speakers make a call here.  To me, "streaming table" sounds like a sentence with more obvious meaning than "stream table".  "stream table" sounds weird to me, like two nouns in the same sentence leaving me in a "What?" state.

Reply:
He didn't agree, but I did it anyway.